### PR TITLE
minor tweaks to dialog [not to merge yet]

### PR DESCRIPTION
**Pull request by TurBoss**
I'm checking this:

http://www.pygtk.org/pygtk2tutorial/sec-Dialogs.html

you are currently using a window?

I don't have glade ( need to fix compilation of 3.8.5)

### CODE REVIEW

**Comment by KurtJacobson** [comment]
Yes, I am using a window, but it should probably be a popup, I will change that.

Once gtk.main_quit() is called wont the window be destroyed anyway since it will no longer have a reference? I don't know what gtk.main really does, so I may well be wrong here.

**Comment by KurtJacobson** [comment]
Are you having trouble with github not loading properly? I can barely use it here

**Comment by TurBoss** [comment]
is working fine for me

I'm working with the dialogs will use window as seems reasonable

**Comment by KurtJacobson** [comment]
OK, must be on my end then. my computer is very old and slow :(

I was thinking about defining global dialog types in at the beginning of dialogs.py

YES_NO = 0
OK_CANCEL = 1
ERROR = 2

Would this be good? Should they be in there own class, or how would they be accessed in hazzy.py?

Thanks!

**Comment by TurBoss** [comment]
I would create only one class and initialize it with the dialog_type parameter

```

dialog_0 = Dialogs(YES_NO)
dialog_1 = Dialogs(OK_CANCEL)
dialog_error = Dialogs(ERROR)

dialog_error.show(message)
```

like we do with the touchpad

do you agree?


another thing

I'm using a callback instead of returning True or False 

how do you see this?

I'm pushing a commit that shows this
https://github.com/KurtJacobson/hazzy/pull/21/commits/05f26346e47499841335f9606593e3b29b619e37#diff-eb4840aa5bac394becb2b7b9579df29dR61

File: hazzy/modules/dialogs/dialogs.py [hunk 1]
```diff
@@ -58,17 +54,16 @@ def __init__(self, message, dialog_type=0):
             self.builder.get_object("button1").set_label("OK")
             self.builder.get_object("button2").set_label("CANCEL")
 
+        self.callback = None
+
     def on_button1_clicked(self, widget, data=None):
-        self.result = True
-        gtk.main_quit()
+        self.window.hide()
+        self.callback()
```
Comment (TurBoss) [inline hazzy/modules/dialogs/dialogs.py]: this calls close_window() from hazzy.py

Comment (TurBoss) [reply hazzy/modules/dialogs/dialogs.py]: maybe not a good Idea...

**Comment by KurtJacobson** [comment]
Sorry, I can't  reply on github.

This should work but the problem is that then we always have to have a
callback to use the dialog, which might be inconvenient.

**Comment by TurBoss** [comment]
yes I was thinking the same

**Comment by TurBoss** [comment]
can you open a gitter room for this repo?
whe can talk there

https://gitter.im

**Comment by KurtJacobson** [comment]
I had not heard of that before, but it is a great idea! Let me see what I
can do

**Comment by TurBoss** [comment]
I'm rethinking...

you always want to call a function with the yes/no ok/cancel dialog

eg when you press "no" or "cancel" just hide the dialog but if "yes" or "ok" something has to be fired, the function you passed to the dialog 

"close_window" or whatever you want

**Comment by KurtJacobson** [comment]
Can't get github to respond so I can set up gitter, I'm going to try
restarting, might be a while :(

Oh, just got you latest email on my phone - that is true, I think that
might work

**Comment by TurBoss** [comment]
weh can talk via irc on freenode

I'm TurBoss too

**Comment by KurtJacobson** [comment]
OK started a irc hazzy channel

**Comment by TurBoss** [comment]
I thinks that this is done

please review

Thanks

**Comment by KurtJacobson** [comment]
This is a great improvement. It used to be that I could get multiple dialogs to pup up because my machine was so slow, but now since you are just showing the one window instead of creating another one this cant happen any more!

File: hazzy/hazzy.py [hunk 2]
```diff
@@ -107,38 +107,47 @@
 
 # Throw up a dialog with debug info when an error is encountered
 def excepthook(exc_type, exc_value, exc_traceback):
+
+    error_dialog = Dialogs(2)
```
Comment (KurtJacobson) [inline hazzy/hazzy.py:111]: Should not `error_dialog = Dialogs(2)` be defined outside of this method so it does not create a new dialog instance each it is called?

**Review by KurtJacobson — APPROVED** [review]

File: hazzy/hazzy.py (hunk 2 above)
Comment (TurBoss) [reply hazzy/hazzy.py:111]: hmmmm

the question is, where should the exceptionhook go?

havent worked with this

**Comment by TurBoss** [comment]
Thanks

**Pull request merged**